### PR TITLE
E2E DualStack Test

### PR DESCRIPTION
**Pull request by dereknola**
#### Proposed Changes ####
- New DualStack E2E test, yaml resources
- Dump vagrant log on cluster creation failure
- Update RunCmdOnNode to add quotes

#### Types of Changes ####
New Test

#### Verification ####
`go test -v -timeout=20m ./tests/e2e/dualstack/... -run E2E`

#### Linked Issues ####
N/A Internal testing change

#### Further Comments ####
Sorry for all the commits, this as been an off and on again process for several weeks. Will squash all when merging.

### CODE REVIEW

File: tests/e2e/dualstack/Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    vm.provision :rke2, run: 'once' do |rke2|
+      rke2.env = %W[INSTALL_RKE2_TYPE=agent #{install_type}]
+      rke2.config_mode = '0644' # side-step https://github.com/k3s-io/k3s/issues/4321
+      rke2.install_path = false
+      rke2.config = <<~YAML
+        write-kubeconfig-mode: 0644
+        node-external-ip: #{node_ip4},#{node_ip6}
+        server: https://#{NETWORK4_PREFIX}.100:9345
+        token: vagrant-rke2
+        cni: calico
```
Comment (manuelbuil) [inline tests/e2e/dualstack/Vagrantfile:87]: The cni config is not needed in the agent config

File: tests/e2e/dualstack/Vagrantfile [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+      rke2.config_mode = '0644' # side-step https://github.com/k3s-io/k3s/issues/4321
+      rke2.config = <<~YAML
+        write-kubeconfig-mode: 0644
+        node-external-ip: #{node_ip4},#{node_ip6}
+        node-ip: #{node_ip4},#{node_ip6}
+        server: https://#{NETWORK4_PREFIX}.100:9345
+        token: vagrant-rke2
+        cluster-cidr: 10.42.0.0/16,2001:cafe:42:0::/56
+        service-cidr: 10.43.0.0/16,2001:cafe:42:1::/112
+        cni: calico
```
Comment (manuelbuil) [inline tests/e2e/dualstack/Vagrantfile:73]: Why did you choose calico?

File: tests/e2e/dualstack/Vagrantfile (hunk 1 above)
Comment (dereknola) [reply tests/e2e/dualstack/Vagrantfile:87]: I'm going to be adding support for all 3 CNIs so it will need to be explicit.

Comment (manuelbuil) [reply tests/e2e/dualstack/Vagrantfile:87]: Ok!

File: tests/e2e/dualstack/Vagrantfile (hunk 2 above)
Comment (manuelbuil) [reply tests/e2e/dualstack/Vagrantfile:73]: Ok. It should work now :D

**Review by manuelbuil — APPROVED** [review]

File: tests/e2e/dualstack/Vagrantfile (hunk 1 above)
Comment (brandond) [reply tests/e2e/dualstack/Vagrantfile:87]: To Manuel's point, agents don't support the CNI flag; it's pulled down from the server config when the agent bootstraps.

**Review by brandond — APPROVED** [review]
LGTM other than the one nit about --cni not being a valid option for agents.

**Pull request merged**